plugins/plugin-java-debugger/che-plugin-java-debugger-server/src/main/java/org/eclipse/che/plugin/jdb/server/JavaDebugger.java

```diff
@@ -361,48 +361,50 @@ public void resume(ResumeAction action) throws DebuggerException {
-            try {
-                variables = currentFrame.getLocalVariables();
-            } catch (DebuggerAbsentInformationException e) {
-                existInformation = false;
-            }
-            for (JdiField f : currentFrame.getFields()) {
-                List<String> variablePath = asList(f.isStatic() ? "static" : "this", f.getName());
-                dump.getFields().add(newDto(FieldDto.class).withIsFinal(f.isFinal())
-                                                           .withIsStatic(f.isStatic())
-                                                           .withIsTransient(f.isTransient())
-                                                           .withIsVolatile(f.isVolatile())
-                                                           .withName(f.getName())
-                                                           .withExistInformation(existInformation)
-                                                           .withValue(f.getValue().getAsString())
-                                                           .withType(f.getTypeName())
-                                                           .withVariablePath(newDto(VariablePathDto.class).withPath(variablePath))
-                                                           .withPrimitive(f.isPrimitive()));
-            }
-            for (JdiLocalVariable var : variables) {
-                dump.getVariables().add(newDto(VariableDto.class).withName(var.getName())
-                                                                 .withExistInformation(existInformation)
-                                                                 .withValue(var.getValue().getAsString())
-                                                                 .withType(var.getTypeName())
-                                                                 .withVariablePath(
-                                                                         newDto(VariablePathDto.class)
-                                                                                 .withPath(singletonList(var.getName()))
-                                                                 )
-                                                                 .withPrimitive(var.isPrimitive()));
-            }
-            return dump;
+            return getCurrentFrame();
         } finally {
             lock.unlock();
         }
     }
 
+    @Override
+    public StackFrameDump getStackFrameDump(long threadId, int frameIndex) throws DebuggerException {
+        lock.lock();
+        try {
+            return new JdbStackFrame(getJdiStackFrame(threadId, frameIndex));
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    @Override
+    public List<ThreadDump> getThreadDumps() throws DebuggerException {
+        List<ThreadDump> threadDumps = new LinkedList<>();
+
+        for (ThreadReference t : vm.allThreads()) {
+            List<JdbStackFrame> frames = new LinkedList<>();
+            try {
```

Comment:
Look,please, com.sun.di.ThreadReference#isSuspended() I think we should use this method instead of using try-catch in this case. More http://www.docjar.com/docs/api/com/sun/jdi/ThreadReference.html#jdoc.